include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -2054,6 +2086,25 @@ macromap<ContextT>::is_valid_concat(string_type new_value,
     return 1 == rescanned.size();
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Bulk update expand positions
+//
+///////////////////////////////////////////////////////////////////////////////
+// batch update tokens with a single expand position
+template <typename ContextT>
+template <typename ContainerT>
+void macromap<ContextT>::set_expand_positions(ContainerT &tokens, position_type pos)
+{
+    typename ContainerT::iterator ex_end = tokens.end();
+    for (typename ContainerT::iterator it = tokens.begin();
+         it != ex_end; ++it) {
+        // expand positions are only used for __LINE__, __FILE__, and macro names
+        if (token_id(*it) == T_IDENTIFIER)
+            it->set_expand_position(pos);
+    }
+}
+
```

Comment:
Will all tokens receive the same position information? Should they receive the corresponding column information as well?

Reply:
IMO, "no" on both. From my investigation, only `__LINE__` and `__FILE__` need to make use of the "position where expanded" information. For implementation reasons general identifiers do as well, in case they are queued instead of immediately expanded - they must pass their own expand positions for use by any `__LINE__` or `__FILE__` macros in their own expansion. That's one of the problems with `BAR` in the test case that triggered this project - it is a function-like macro whose name token and arguments come from separate macro expansions.

As for the columns, I haven't seen any cases where the actual column number is needed. The `expand_pos` fields do capture it, but we don't use them.

Reply:
@jefftrull thanks for the explanations, makes sense. As for columns, they are used for error messages, I believe.